### ISSUE TITLE
Requires should be separated by newlines

### ISSUE THREAD

**Issue by pesterhazy** [kkinnear/zprint#2]
Great project! I've built [boot-fmt](https://github.com/pesterhazy/boot-fmt), similar to `lein-zprint`. It can be used with lein or boot equally.

In dogfooding the auto-formatting feature in my core.clj, I noticed that zprint formats requires to go in a single line:

https://github.com/pesterhazy/boot-fmt/blob/master/src/boot_fmt/core.clj#L4

IMO `(:requires)` should always be separated by newlines, even if they are short. What do you think?

**Comment by kkinnear** [comment]
Wow, boot-fmt -- go you!  That's great!  I haven't given it a try yet, but I will soon.  Sorry it took me a while to get back to you on this, I've been pretty busy since I returned from the conj.

One of the things about zprint is that, by default, it puts _almost_ anything that will fit on a single line on a single line, since the basic point was to fit as much into the available space while looking as good as possible.  Most of my requires are forced multi-line just because of their length.  That said, I can make `:require` force new lines by adding it to the `:fn-map`, thus:
```clojure
zprint.zprint=> rq
"(:require [boot-fmt.impl :as impl] [boot.core :as bc] [boot.util :as bu])"
zprint.zprint=> (czprint rq {:parse-string? true})
(:require [boot-fmt.impl :as impl] [boot.core :as bc] [boot.util :as bu])
nil
zprint.zprint=> (czprint rq {:parse-string? true :fn-map {":require" :force-nl-body}})
(:require [boot-fmt.impl :as impl]
          [boot.core :as bc]
          [boot.util :as bu])
nil
```
I'm thinking that this is more what you had in mind?  Probably everybody would like that, so if you agree, I'll add `:require` to the `:fn-map` with `:force-nl-body` as its function type.

Thanks again for boot-fmt!

**Comment by mynomoto** [comment]
@kkinnear Is there a config for formating `require` as:
```clojure
(:require
  [boot-fmt.impl :as impl]
  [boot.core :as bc]
  [boot.util :as bu])
```
I'm currently using your solution above which works great but it's not my favorite style :nail_care:

Related to config in general, I'm more concerned about uniformity than with more code in a screen so I'm adding lots of things on `:fn-force-nl` but that doesn't work for all types of functions right? Is the rule for it to work only for `-nl` function types? It currently works for `:arg1` but not `:arg1-body1` ones.

**Comment by kkinnear** [comment]
Good questions (the second resulting in my discovering two bugs as well)!

1. There actually isn't a way to get the `:require` to print as you suggest.  I will add a `fn-type` which will force a flow for the arguments regardless of whether they would hang or not.   I don't know just what to call it yet.

2. You may have already discovered that if you change `:fn-force-nl` you need to specify the entire new set, not an addition to the set.  That's a bug, and I'll fix it (and it may break your configuration, so be warned).  

3. Regarding the question about `:arg1` and `:arg1-body` -- what actually happens is that all of the `-body` `:fn-type`s are noticed, and the indent is set, and then the `-body` is stripped off.  When they get matched against the set from `:fn-force-nl`, they don't have `-body` anymore.  So, if you put `:arg1-body` in `:fn-force-nl`, you get nothing. (You might notice that I also didn't realize this when I populated the `:fn-force-nl` set.)  But if you put `:arg1` in `:fn-force-nl`, you cover both `:arg1` and `:arg1-body`.  Thus:

```clojure
zprint.zprint=> is
"(if this? (that and the other) (and more and more))"
zprint.zprint=> ffn
#{:force-nl :noarg1 :noarg1-body :force-nl-body :arg1-force-nl}
zprint.zprint=> (czprint is {:parse-string? true :fn-force-nl (into #{} (concat ffn #{:arg1}))})
(if this?
  (that and the other)
  (and more and more))
```
So, if you put `:arg1` in `:fn-force-nl`, you get both `:arg1` and `:arg1-body` for free.  Which isn't ideal, actually.  I think that it should only affect the ones that are actually in the set.  So I'll fix that too.  

That said, as long as you don't mind picking up `:arg1` functions as well as `:arg1-body` functions, you should be good to go for now.

**Comment by mynomoto** [comment]
Yeah, I figured having to configure the whole `:fn-force-nl` set. Thanks for the explanation about `-body`, it's quite helpful. I'm using `:arg1` for now and it works fine.

Thank you so much for this library! I have other suggestions but I'll stop hijacking this issue and open a new one.

**Comment by kkinnear** [comment]
This is fixed in 0.2.10.  

I ended up doing this by default:
```clojure
(:require [boot-fmt.impl :as impl]
          [boot.core :as bc]
          [boot.util :as bu])
```
but if you define `:require` as `:flow-body`, it will do this:
```clojure
(:require
  [boot-fmt.impl :as impl]
  [boot.core :as bc]
  [boot.util :as bu])
```
if that is your preference.

Thanks to you both for the suggestions!

**Comment by pesterhazy** [comment]
Thanks for the new release, it works like a charm! I pushed out a new release of boot-fmt with the changes.

This allowed me to reduce my dogfooding zprint config to just `:style :community`: https://github.com/pesterhazy/boot-fmt/commit/47e94b42d45a9323b976b9b8f37edd29adc72888
